cpp_ext/TorchValues.h

```diff
@@ -136,11 +137,28 @@ class PyTorch_NoneValue : public PyConcreteValue<PyTorch_NoneValue> {
   };
 DECLARE_SCALAR_VALUE(Bool, bool, bool)
 DECLARE_SCALAR_VALUE(Device, int, device)
-DECLARE_SCALAR_VALUE(Int, int, int)
 DECLARE_SCALAR_VALUE(Float, float, float)
 DECLARE_SCALAR_VALUE(String, std::string, str)
 #undef DECLARE_SCALAR_VALUE
 
+class PyTorch_IntValue : public PyConcreteValue<PyTorch_IntValue> {
+public:
+  static constexpr IsAFunctionTy isaFunction = isATorch_IntValue;
+  static constexpr const char *pyClassName = "Torch_IntValue";
+  using PyConcreteValue::PyConcreteValue;
+  static void bindDerived(ClassTy &c);
+  PyTorch_IntValue(int b)
+      : PyTorch_IntValue(mlir::python::PyGlobals::get()
+                             .lookupOperationClass("torch.constant.int")
+                             .value()(b)
+                             .cast<PyTorch_IntValue>()) {}
+  PyTorch_IntValue(DType b)
+      : PyTorch_IntValue(mlir::python::PyGlobals::get()
+                             .lookupOperationClass("torch.constant.int")
+                             .value()(to_underlying(b))
```

Comment:
the `to_underlying` is the only non-boiler plate part of the whole PR - https://stackoverflow.com/a/33083231